app/com/linkedin/drelephant/spark/SparkMetricsAggregator.scala

```diff
@@ -83,16 +84,25 @@ class SparkMetricsAggregator(private val aggregatorConfigurationData: Aggregator
     }
   }
 
-  private def aggregateresourcesActuallyUsed(executorMemoryBytes: Long, totalExecutorTaskTimeMillis: BigInt): BigInt = {
-    val bytesMillis = BigInt(executorMemoryBytes) * totalExecutorTaskTimeMillis
-    (bytesMillis / (BigInt(FileUtils.ONE_MB) * BigInt(Statistics.SECOND_IN_MS)))
+  //calculates the resource usage by summing up the resources used per executor
+  private def calculateResourceUsage(executorSummaries: Seq[ExecutorSummary]): BigInt = {
+    var sumResourceUsage: BigInt = 0
+    executorSummaries.foreach(
+      executorSummary => {
+        var memUsed: Long = executorSummary.peakJvmUsedMemory.getOrElse(JVM_USED_MEMORY, 0) //+ MemoryFormatUtils.stringToBytes("300M")
```

Comment:
We might want to add a buffer on top of the peak memory.

Reply:
The variable "resourcesActuallyUsedWithBuffer" (line no: 64) is the resource usage with buffer and for calculating resources wasted, this value is being considered.